flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -258,13 +456,34 @@ public void createTable(CatalogBaseTable table, ObjectIdentifier objectIdentifie
 			"CreateTable");
 	}
 
+	/**
+	 * Creates a temporary table in a given fully qualified path.
+	 *
+	 * @param table The table to put in the given path.
+	 * @param objectIdentifier The fully qualified path where to put the table.
+	 * @param replace controls what happens if a table exists in the given path,
+	 *                if true the table is replaced, an exception will be thrown otherwise
+	 */
+	public void createTemporaryTable(
+			CatalogBaseTable table,
+			ObjectIdentifier objectIdentifier,
+			boolean replace) {
+		temporaryTables.compute(objectIdentifier, (k, v) -> {
+			if (v != null && !replace) {
+				throw new ValidationException(String.format("Temporary table %s already exists", objectIdentifier));
+			} else {
+				return table;
+			}
+		});
+	}
+
 	/**
 	 * Alters a table in a given fully qualified path.
 	 *
 	 * @param table The table to put in the given path
 	 * @param objectIdentifier The fully qualified path where to alter the table.
 	 * @param ignoreIfNotExists If false exception will be thrown if the table or database or catalog to be altered
-	 * does not exist.
+	 *                          does not exist.
```

Comment:
minor: revert

Reply:
This was requested by @twalthr to indent long javadocs